sympy/core/add.py

```diff
@@ -593,9 +593,11 @@ def _eval_is_positive(self):
 
     def _eval_is_nonnegative(self):
         from sympy.core.exprtools import _monotonic_sign
+        from sympy import simplify
+        self = simplify(self)
         if not self.is_number:
             c, a = self.as_coeff_Add()
-            if not c.is_zero and a.is_nonnegative:
```

Comment:
```as_coeff_Add()``` function return ```1/x``` value--->``` a  = _pos``` and ```x``` value---> ```a =x```.
if we make condition if ```a = variable``` then we not need to use simplify and if ```a=_pos/_neg``` in that case we use simplify.it's solve also ```trigonometry formula``` as problem in above example.